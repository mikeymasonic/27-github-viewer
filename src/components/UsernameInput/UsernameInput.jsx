import React from 'react';
import PropTypes from 'prop-types';

const UsernameInput = ({ username, onUserChange }) => (
  <>
    <input type="text" name="username" value={username} onChange={onUserChange} />
  </>
);

UsernameInput.propTypes = {
  username: PropTypes.string.isRequired,
  onUserChange: PropTypes.func.isRequired
};

export default UsernameInput;
