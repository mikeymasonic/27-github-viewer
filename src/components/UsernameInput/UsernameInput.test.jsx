import React from 'react';
import { shallow } from 'enzyme';
import UsernameInput from './UsernameInput';

describe('UsernameInput component', () => {
  it('renders UsernameInput', () => {
    const wrapper = shallow(<UsernameInput 
      username="mikeymasonic"
      onUsernameChange={() => {}}
      onUsernameSubmit={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
