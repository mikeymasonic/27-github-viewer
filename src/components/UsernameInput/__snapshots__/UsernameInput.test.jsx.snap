// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsernameInput component renders UsernameInput 1`] = `
<Fragment>
  <input
    name="username"
    onChange={[Function]}
    type="text"
    value="mikeymasonic"
  />
  <input
    name="search"
    onChange={[Function]}
    type="button"
    value="Search"
  />
</Fragment>
`;
